internal/controller/secrets/secrets_test.go

```diff
@@ -0,0 +1,296 @@
+	kubernetesfake "k8s.io/client-go/kubernetes/fake"
+	kubetesting "k8s.io/client-go/testing"
+
+	"go.pinniped.dev/internal/controllerlib"
+	"go.pinniped.dev/internal/plog"
+	"go.pinniped.dev/internal/testutil"
+)
+
+func TestNewServiceAccountTokenCleanupController(t *testing.T) {
+	namespace := "a-namespace"
+	legacySecretName := "a-secret"
+	observableWithInformerOption := testutil.NewObservableWithInformerOption()
+	secretsInformer := kubeinformers.NewSharedInformerFactory(nil, 0).Core().V1().Secrets()
+
+	var log bytes.Buffer
+	_ = NewServiceAccountTokenCleanupController(
+		namespace,
+		legacySecretName,
+		nil, // not needed for this test
+		secretsInformer,
+		observableWithInformerOption.WithInformer,
+		plog.TestLogger(t, &log),
+	)
+
+	secretsInformerFilter := observableWithInformerOption.GetFilterForInformer(secretsInformer)
+
+	legacySecret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: legacySecretName, Namespace: namespace}, Type: corev1.SecretTypeServiceAccountToken}
+	wrongName := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "wrongName", Namespace: namespace}, Type: corev1.SecretTypeServiceAccountToken}
+	wrongType := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "wrongType", Namespace: namespace}, Type: "other-type"}
+	wrongNamespace := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "wrongNamespace", Namespace: "wrong-namespace"}, Type: corev1.SecretTypeServiceAccountToken}
+	wrongObject := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "config-map", Namespace: namespace}}
+
+	require.False(t, secretsInformerFilter.Add(wrongName))
+	require.False(t, secretsInformerFilter.Update(wrongName, wrongNamespace))
+	require.False(t, secretsInformerFilter.Update(wrongNamespace, wrongName))
+	require.False(t, secretsInformerFilter.Delete(wrongName))
+
+	require.False(t, secretsInformerFilter.Add(wrongObject))
+	require.False(t, secretsInformerFilter.Update(wrongObject, wrongNamespace))
+	require.False(t, secretsInformerFilter.Update(wrongNamespace, wrongObject))
+	require.False(t, secretsInformerFilter.Delete(wrongObject))
+
+	require.False(t, secretsInformerFilter.Add(wrongNamespace))
+	require.False(t, secretsInformerFilter.Update(wrongNamespace, wrongObject))
+	require.False(t, secretsInformerFilter.Update(wrongObject, wrongNamespace))
+	require.False(t, secretsInformerFilter.Delete(wrongNamespace))
+
+	require.False(t, secretsInformerFilter.Add(wrongType))
+	require.False(t, secretsInformerFilter.Update(wrongType, wrongNamespace))
+	require.False(t, secretsInformerFilter.Update(wrongNamespace, wrongType))
+	require.False(t, secretsInformerFilter.Delete(wrongType))
+
+	require.True(t, secretsInformerFilter.Add(legacySecret))
+	require.True(t, secretsInformerFilter.Update(legacySecret, wrongNamespace))
+	require.True(t, secretsInformerFilter.Update(wrongNamespace, legacySecret))
+	require.True(t, secretsInformerFilter.Delete(legacySecret))
+}
+
+func TestSync(t *testing.T) {
```

Comment:
Is it worth converting the Sync tests to be one test with a test table? That would make it share more code, and easier to change in the future.

Reply:
Yes. This should block merging the PR.

Reply:
This was done.